### ISSUE TITLE
Package request: OrcaSlicer

### ISSUE THREAD

**Issue by MohrJonas** [NixOS/nixpkgs#239390]
**Project description**
Orca Slicer is a slicer for 3d printers, originally forked from Bambu Studio,

**Metadata**

* homepage URL: https://github.com/SoftFever/OrcaSlicer
* source URL: https://github.com/SoftFever/OrcaSlicer
* license: AGPL-3.0
* platforms: linux, darwin

**Comment by taurr** [comment]
Would be really nice to get this one - actually a must in order to use NixOS on my main.

Tried using the `DockerBuild.sh` and `DockeeRun.sh` that are part of OrcaSlicer - and even though this is how I use it under Manjaro, it doesn't work here... It builds but doesn't run.

**Comment by Wulfsta** [comment]
I'll take a crack at it today, no guarantees.

**Comment by Wulfsta** [comment]
Hit an issue with linking, haven't been able to figure out what it is.
```
[100%] Linking CXX executable orca-slicer
/nix/store/r5x7gyddzwml0h9g7gg7r6zlwpdzc10a-binutils-2.40/bin/ld: cannot find -lcereal: No such file or directory
collect2: error: ld returned 1 exit status
make[2]: *** [src/CMakeFiles/OrcaSlicer.dir/build.make:186: src/orca-slicer] Error 1
make[1]: *** [CMakeFiles/Makefile2:522: src/CMakeFiles/OrcaSlicer.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
error: boost::bad_format_string: format-string is ill-formed
```
Keep an eye on #206495, one that is in it will be much easier to follow up with OrcaSlicer. If I can get this issue figured out I'll make a flake for it.

**Comment by ovlach** [comment]
Try https://github.com/ovlach/nix-orca-slicer ... I based it on zhaofengli bambu-studio PR, and seems work on my nix.

**Comment by Wulfsta** [comment]
I see a segfault on my end:
```
[2023-07-23 17:16:28.702473] [0x00007fc981955c80] [trace]   Initializing StaticPrintConfigs
add font of HarmonyOS_Sans_SC_Bold returns 1
add font of HarmonyOS_Sans_SC_Regular returns 1
zsh: segmentation fault (core dumped)  orca-slicer
```

**Comment by ovlach** [comment]
> I see a segfault on my end:
> 
> ```
> [2023-07-23 17:16:28.702473] [0x00007fc981955c80] [trace]   Initializing StaticPrintConfigs
> add font of HarmonyOS_Sans_SC_Bold returns 1
> add font of HarmonyOS_Sans_SC_Regular returns 1
> zsh: segmentation fault (core dumped)  orca-slicer
> ```

Seems some problem with initialize configs. If i use old configs (from differently compiled version), works fine. If i delete configs, ends with segv.

**Comment by ovlach** [comment]
@Wulfsta Can you try-it again (nix run github:ovlach/nix-orca-slicer#orca-slicer --refresh) --  without update flake. I have added a fix from related issue.

I suspect this is an issue with cereal, but am unsure?
Yes, seems it's broken on some version of nixpkgs (i will investigate-it in future, but now runs with locked flake)...

**Comment by Wulfsta** [comment]
@ovlach seems to be building - I'll set it up and test it out a bit. Thanks for the work, if you plan on adding this in a future PR feel free to ping me for review.

**Comment by Wulfsta** [comment]
I think reverting `fa9859507ba72b72f157b42da4ef0cc89a082074` should fix it?

**Comment by ovlach** [comment]
@Wulfsta Hi, i try update nixpkgs defined in flake.lock
cannot find -lcereal - on gentoo site which you mention at cereal upgrade commit was link to prusa slicer commit. I apply patch and now works.
there is also some new issue with some ambigous calls -- also fixed.

if you plan on adding this in a future PR feel free to ping me for review. - yes of course, but i will wait until #206495 will be merged.

**Comment by aacebedo** [comment]
I got a running package based on the app image. However I got some strange Javascript error at startup and icon theme in open file dialog is broken. 

I am trying now to build it form source but I have issues with wxGTK32 which do not include the wx_webkit library by default. It seems to be linked to Mac support only but I don't find any explanation why.

**Comment by ovlach** [comment]
> I got a running package based on the app image. However I got some strange Javascript error at startup and icon theme in open file dialog is broken.
> 
> I am trying now to build it form source but I have issues with wxGTK32 which do not include the wx_webkit library by default. It seems to be linked to Mac support only but I don't find any explanation why.

PrusaSlicer and forks (included BambuStudio / OrcaSlicer) is compatible with wxGTK31 (best compatible). You need enable this https://github.com/ovlach/nix-orca-slicer/blob/main/wxGTK31.nix#L92-L93 . Also you will need some patches from original PrusaSlicer and some patches to ensure compatibility with new version of g++/gcc and libcereral (i make somes PR to OrcaSlicer, so some patches is already included in beta i think)

**Comment by aacebedo** [comment]
Ok I'll try to check this out ASAP

**Comment by ovlach** [comment]
@aacebedo waiting until https://github.com/NixOS/nixpkgs/pull/206495 merged. wxGTK31.nix patches is from bambu-studio PR.

**Comment by Wulfsta** [comment]
@ovlach mind updating this to 1.7.0? I'd like to try out the Prusa Organic Supports.

**Comment by ovlach** [comment]
@Wulfsta Yes, but now I'm on vacation in Egypt (and don't have a machine with me to try it), so feel free to make a PR, or you will have to wait until 10-12 :-/.

**Comment by Wulfsta** [comment]
It is unlikely I will have time to work around the build issue I encountered before then (though admittedly I haven’t looked into it), enjoy your vacation!

**Comment by blkgoose** [comment]
I'm having issues after trying to login with google SSO, I can see the log trying to use the access token, and after a couple of seconds it just SIGSEGV.

I even tried using the username and password system, it logs me in, and after a couple of seconds it SIGSEGV again, and it continue doing so after every restart.

**Comment by ovlach** [comment]
> @ovlach mind updating this to 1.7.0? I'd like to try out the Prusa Organic Supports.
Version is updated to 1.7.0

**Comment by ovlach** [comment]
> @ovlach mind updating this to 1.7.0? I'd like to try out the Prusa Organic Supports.

@Wulfsta Version is updated to 1.7.0

**Comment by Janik-Haag** [comment]
@ovlach would you consider opening a PR to nixpkgs adding orcaslicer, I would gladly review it.

**Comment by aacebedo** [comment]
Is it still working? On my side since a few weeks it segfault. Same behavior on BambuStudio.

**Comment by tylergets** [comment]
I was able to run the latest beta by wrapping the app image
```
{ appimageTools
, fetchurl
, lib
, pkgs
}:

appimageTools.wrapType2 rec {
  pname = "bambu-studio";
  version = "4.5.0";

  src = fetchurl {
    url = "https://github.com/bambulab/BambuStudio/releases/download/v01.08.00.62/BambuStudio_linux_ubuntu_v01.08.00.62-20231123075101.AppImage";
    sha256 = "sha256-4JWtQ3NkkTv5I6aF7kULfqrgBHffy7aA+iG+3oK2OnE=";
  };

  profile = ''
    export LC_ALL=C.UTF-8
  '';

  multiPkgs = extraPkgs;
  extraPkgs =
    pkgs: (appimageTools.defaultFhsEnvArgs.multiPkgs pkgs) ++ ([
      pkgs.webkitgtk
      pkgs.gst_all_1.gstreamer
      pkgs.gst_all_1.gst-libav
      pkgs.gst_all_1.gst-plugins-bad
    ]);

  meta = with lib; {
    homepage = "https://github.com/bambulab/BambuStudio";
    description = "A prusa slicer fork";
    platforms = [ "x86_64-linux" ];
    license = "unknown";
  };
}

```

**Comment by Wulfsta** [comment]
@ovlach bambu-studio has been merged, you should be unblocked on creating a PR; let me know if you do.

**Comment by Wulfsta** [comment]
Anyone looking to use the v1.8.1 OrcaSlicer release can simply use [this](https://github.com/Wulfsta/orca-slicer-flake) in the interim.

**Comment by pinpox** [comment]
@Janik-Haag  I opened a PR based on @Wulfsta 's flake here: https://github.com/NixOS/nixpkgs/pull/275638
Mostly copy/pasting bambu-studio package, let me know if can be simplified or deduplicated

**Comment by Janik-Haag** [comment]
I'll review it once I find time ^^

**Comment by Wulfsta** [comment]
Closed by #275638
